adr/ADR-15.md

```diff
@@ -0,0 +1,138 @@
+
+If there was a consumer lookup but an error occurs, the behavior depends on few things:
+
+- If the error is a "not found" (that is, the library got a response indicating that the said consumer does not exists):
+    - If the subscription is "bound" (stream and consumer names were provided), the error is returned.
+    - If the subscription is not "bound", then it is ok and the library will attempt to create the JS consumer.
+- For a "lookup error", for instance because the consumer info API is not allowed, or the consumer is not in the current account, etc.. then unless this is a Pull subscription, the library should return an error.
+
+#### NATS subscription
+
+At this point the NATS subscription will be created on the deliver subject found from the consumer info or on an inbox.
+
+Note that the subscription should be created prior to attempt to create the JS consumer (when applicable) because for durable subscriptions, this is the only way for the server to detect if the durable is already in use or not. If there is no subscription (no interest), then the server will simply update the delivery subject of the JS durable consumer.
+
+#### Creating the JS Consumer
+
+If the library determined that it should attempt to create a JS consumer (no consumer name provided, or durable name that does not exists), the library will fill a consumer config with what was provided by the user, but add the `DeliverSubject` to the inbox (but leave it blank for pull subscription), set the `FilterSubject` to the user provided subject, ensure that `AckPolicy` is set and possibly some other fields such as `MaxAckPending`.
+
+The request is then sent to `<JS prefix>.CONSUMER.DURABLE.CREATE.<stream name>.<durable name>` for a durable subscription or `<JS prefix>.CONSUMER.CREATE.<stream name>` for an ephemeral.
+
+If the result of this request is "OK", it means that the server has created the JS consumer and uses the deliver subject the NATS subscription is created on. For ephemeral, the consumer name will be saved from the creation request's response, so that the ephemeral JS consumer can be deleted on Drain and Unsubscribe.
+
+If the result indicates that the consumer already exists, then it means that there is an active durable consumer, with some NATS subscription attached and so we should fail this, *unless* a Queue name was provided. The client/server currently does not use the concept of a queue group in the JS consumer, but if the server is enhanced, as part of getting the consumer information, the library will be able to determine if the consumer that already exists is part of the same group, in which case it is fine, or not, in which case an error should be returned.
+
+When the consumer already exists, the NATS queue subscription that was created prior to the "AddConsumer" call needs to be destroyed and replaced with the new NATS queue subscription on the consumer info's `DeliverSubject`.
+
+### Runtime
+
+#### Heartbeats
```

Comment:
we need to update for recent discussions that we should always handle them even if we think they shouldnt be coming or we jsut dont know.

Maybe worth mentioning about FC with details in the next section